File: VCF/subsetSNPbyBED/subsetSNPbyBED.m
function [] = subsetSNPbyBED(snpdir,bedfile,flank,maf,vt,outdir)
if (exist(outdir)==0)
	system(horzcat('mkdir ',outdir));
end
[chr,spos,epos,~,~,~] = textread(bedfile,'%s %s %s %s %s %s');

uni_chr = unique(chr);
for chridx=1:length(uni_chr)
	thischr = uni_chr{chridx};
	display(horzcat('subseting ',thischr));
	snpfile = horzcat(snpdir,'/',thischr,'.vcf');
	if (exist(snpfile)==0)
		continue;
	end
	outfile = horzcat(outdir,'/',thischr,'.vcf');
	
	pick = find(ismember(chr,thischr));
                                                                              
    tempfile = horzcat(thischr,'.temp');
    out = fopen(tempfile,'w');
	for i = 1:length(pick)
		fprintf(out,'%s\t%s\t%s\n',chr{pick(i)},spos{pick(i)},epos{pick(i)});
	end
	fclose(out);
	
	if (exist(outfile)~=0)
		system(horzcat('rm ',outfile));
	end
	cmd = horzcat('python subsetSNPbyBED.py ',snpfile,' ', tempfile,' ',num2str(flank),' ',num2str(maf),' ',vt,' >> ',outfile);
	system(cmd);
	system(horzcat('rm ',tempfile));
end

File: VCF/subsetSNPbyBED/subsetSNPbyBED.R
args = commandArgs(T)
VCFdir = args[1]
bedfile = args[2]
flank = args[3]
outdir = args[4]

re = dir.create(outdir, showWarnings = FALSE,recursive=T)
if (!re){
	print('Outdir exists! will be emptied')
	system(paste0('rm -r ',outdir))
	system(paste0('mkdir ',outdir))
}
mat <- read.delim(bedfile,header=F,stringsAsFactors=F)

files <- list.files(VCFdir,pattern = "\\.vcf$")
for (fileidx in 1:length(files)){
	file = files[fileidx]
	thischr = strsplit(file,split='.vcf')[[1]][1]
	print(paste0('subseting ',thischr));
	
	outfile = paste0(outdir,'/',thischr,'.vcf');
	
	pick = which(mat[,1] == thischr)
	if (length(pick)>0){
		part = mat[pick,]
		tmpfile = tempfile()
		write.table(part,file=tmpfile,row.names=F,col.names=F,quote=F,sep="\t")
		cmd = paste('python /cluster/zeng/code/research/tools/VCF/subsetSNPbyBED/subsetSNPbyBED.py ',file.path(VCFdir,file),tmpfile,flank,'>',outfile,sep=' ');
		system(cmd);
		system(paste0('rm ',tmpfile));
	}
}
